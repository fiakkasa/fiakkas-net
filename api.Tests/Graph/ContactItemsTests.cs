using HotChocolate.Execution;

namespace api.Tests.Graph;

public class ContactItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task ContactItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              contactItems {
                totalCount
                nodes {
                  createdAt
                  description
                  href
                  icon
                  id
                  internalId
                  key
                  title
                  updatedAt
                  version
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task ContactItemById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              contactItemById(id: "Q29udGFjdEl0ZW0KZ2ViZjIyNGE4N2ZmMzQ3Yjk4ODJiZGQ0MWVjN2Y1YTA1") {
                createdAt
                description
                href
                icon
                id
                internalId
                key
                title
                updatedAt
                version
              }
            }
            """);

        var fn = result.ExpectOperationResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
