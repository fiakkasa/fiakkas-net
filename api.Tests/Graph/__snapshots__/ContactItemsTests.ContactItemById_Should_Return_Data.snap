{
  "data": {
    "contactItemById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "description": "Content",
      "href": "/test",
      "icon": "Icon",
      "id": "Q29udGFjdEl0ZW06qCTy6/N/uUeIK91B7H9aBQ==",
      "internalId": "ebf224a8-7ff3-47b9-882b-dd41ec7f5a05",
      "key": "Key",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
